src/core/text/Text.js

```diff
@@ -125,14 +128,15 @@ Object.defineProperties(Text.prototype, {
         {
             this.updateText(true);
 
-            return  this.scale.y * this._texture._frame.height;
+            return Math.abs(this.scale.y) * this._texture.orig.height;
         },
         set: function (value)
         {
             this.updateText(true);
 
-            this.scale.y = value / this._texture._frame.height;
-            this._height = value;
+            var sign = utils.sign(this.scale.x) || 1;
+            this.scale.x = sign * value / this.texture.orig.width;
+            this._width = value;
```

Comment:
Is it ok that we change values of the _width and scale.x properties inside of the height setter? Was it planned this way? It looks like this is a typo and this part of the setter was just copy-pasted from the width setter.